node/nodeBlocks.go

```diff
@@ -1,34 +1,88 @@
 package node
 
 import (
+	"fmt"
+
 	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
 	"github.com/ElrondNetwork/elrond-go/process"
 )
 
-func (n *Node) getBlockHeaderByHash(hash []byte) (data.HeaderHandler, error) {
-	return process.GetHeaderFromStorage(
-		n.processComponents.ShardCoordinator().SelfId(),
-		hash,
-		n.coreComponents.InternalMarshalizer(),
-		n.dataComponents.StorageService(),
-	)
-}
+func (n *Node) getBlockHeaderByNonce(nonce uint64) (data.HeaderHandler, []byte, error) {
+	headerHash, err := n.getBlockHashByNonce(nonce)
+	if err != nil {
+		return nil, nil, err
+	}
 
-func (n *Node) getBlockRootHash(headerHash []byte, header data.HeaderHandler) []byte {
-	blockRootHash, err := n.processComponents.ScheduledTxsExecutionHandler().GetScheduledRootHashForHeader(headerHash)
+	header, err := n.getBlockHeaderByHash(headerHash)
 	if err != nil {
-		blockRootHash = header.GetRootHash()
+		return nil, nil, err
 	}
 
-	return blockRootHash
+	return header, headerHash, nil
 }
 
-func (n *Node) getBlockHeaderByNonce(nonce uint64) (data.HeaderHandler, []byte, error) {
-	return process.GetHeaderFromStorageWithNonce(
+func (n *Node) getBlockHashByNonce(nonce uint64) ([]byte, error) {
+	shardId := n.processComponents.ShardCoordinator().SelfId()
+	hashByNonceUnit := dataRetriever.GetHdrNonceHashDataUnit(shardId)
+
+	return process.GetHeaderHashFromStorageWithNonce(
 		nonce,
-		n.processComponents.ShardCoordinator().SelfId(),
 		n.dataComponents.StorageService(),
 		n.coreComponents.Uint64ByteSliceConverter(),
 		n.coreComponents.InternalMarshalizer(),
+		hashByNonceUnit,
 	)
 }
+
+func (n *Node) getBlockHeaderByHash(headerHash []byte) (data.HeaderHandler, error) {
+	epoch, err := n.getEpochByHash(headerHash)
```

Comment:
can we add some support in case dblookup extensions is not enabled?
so we do not return error for example when we activate API calls on validators, and we query the recent (current epoch) block hashes.

With these changes, for validators the api will always return error (db lookup extension not activated) and previously it was not.

Reply:
Fixed to incorporate a fallback mechanism when `dblookup extensions` is not enabled (theoretically, wasn't a breaking change - the new query parameters have been recently added).